src/types.ts

```diff
@@ -37,7 +37,8 @@ interface NormalizerInput {
 export interface PluginOptions extends GatsbyPluginOptions {
   name: string
   engine: Engine
-  engineOptions?: FlexSearchCreateOptions | LunrPluginOptions
+  engineOptions?: FlexSearchCreateOptions
+  lunrOptions?: any
```

Comment:
I propose a breaking change, but what about merging engine, engineOptions and lunrOptions, and add something like:
```ts
export type Engine =
  | {
    name: 'lunr'
    options: {
      k1?: number
      b?: number
    }
  | {
    name: 'flexssearch'
    options: FlexSearchOptions
  }
```